### PR TITLE
SafeArea helpers

### PR DESCRIPTION
**Pull request by alexpersian**
In iOS 11.0 the new `safeAreaInsets` and `safeAreaLayoutGuide` features were added to handle laying out based on the Safe Area. This feature is not available in iOS 10 and below though, and if you try to use them the layout will go to the `superview` instead of the layout guides. Meaning that you need to make sure to wrap the safe area access in `#available` calls all the time.

This PR adds helper properties to remove the need for wrapping iOS 11.0 checks when setting layout guides.

### CODE REVIEW

File: SwiftWisdom/Core/UIKit/UIView+SafeArea.swift [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            return self.safeAreaInsets
+        } else {
+            return UIEdgeInsets.zero
+        }
+    }
+
+    /// Provides the view's top anchor with regards to the safe area.
+    ///
+    /// **Note:** for iOS 11 and above this anchor will not be available
+    /// until after `viewDidAppear` is called
```
Comment (petey3) [inline SwiftWisdom/Core/UIKit/UIView+SafeArea.swift]: nit: might want to clarify that it is _available_ before `viewDidAppear`, but it is not *accurate* until it has been added to the view hierarchy and `viewDidAppear` is called

File: SwiftWisdom/Core/UIKit/UIView+SafeArea.swift [hunk 2]
```diff
@@ -0,0 +1,71 @@
+//
+//  UIView+SafeArea.swift
+//  SwiftWisdom
+//
+//  Created by Persian, Alex on 11/3/17.
+//  Copyright © 2017 Intrepid. All rights reserved.
+//
+
```
Comment (petey3) [inline SwiftWisdom/Core/UIKit/UIView+SafeArea.swift:8]: Maybe just throw this link in here at the top?
https://developer.apple.com/documentation/uikit/uiview/2891102-safearealayoutguide

**Review by petey3 — APPROVED** [review]
LGTM

**Comment by ghost** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The iPhone simluator specified in your Jenkinsfile is invalid for this version of Xcode</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 167 tests, with 0 failures (0 unexpected) in 0.561 (0.682) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Review by bobgilmore — APPROVED** [review]

**Review by Benuuu — APPROVED** [review]

**Pull request merged**